src/psyclone/parse/file_info.py

```diff
@@ -77,27 +143,286 @@ def filename(self):
+            # cache_updated = True
+            pass
+
+        if not cache_updated:
+            return None
+
+        # Open cache file
+        try:
+            filehandler = open(self._filepath_cache, "wb")
+        except Exception as err:
+            if verbose:
+                print("  - Unable to write to cache file" + str(err))
+            return None
+
+        # Dump to cache file
+        try:
+            pickle.dump(self._cache_data_save, filehandler)
+        except Exception as err:
+            # Invalidate cache
+            self._cache_data_save = None
+            print("Error while storing cache data - ignoring: " + str(err))
+            return None
+
+        if verbose:
+            print(
+                f"  - Cache file updated with "
+                f"hashsum '{self._cache_data_save._source_code_hash_sum}"
+            )
+
+    def get_fparser_tree(
+                self,
+                verbose: bool = False,
+                save_to_cache: bool = True
+            ) -> Fortran2003.Program:
+        """Returns the fparser Fortran2008 representation of the source code.
+
+        :param save_to_cache: Cache is updated if fparser was
+            not loaded from cache.
+        :param verbose: Produce some verbose output
+
+        :returns: fparser representation.
+
+        :raises FileInfoFParserError: if fparser had issues
+        """
+        if self._fparser_tree is not None:
+            return self._fparser_tree
+
+        if verbose:
+            print(f"- Source file '{self._filename}': " f"Running fparser")
+
+        source_code = self.get_source_code()
+        assert self._source_code_hash_sum is not None
+
+        # Check for cache
+        self._cache_load(verbose=verbose)
+
+        if self._cache_data_load is not None:
+            if self._cache_data_load._fparser_tree is not None:
+                if verbose:
+                    print(
+                        f"  - Using cache of fparser tree with hashsum"
+                        f" {self._cache_data_load._source_code_hash_sum}"
+                    )
+
+                # Use cached version
+                self._fparser_tree = self._cache_data_load._fparser_tree
+                return self._fparser_tree
+
+        try:
+            reader = FortranStringReader(
+                source_code, include_dirs=Config.get().include_paths
+            )
+            parser = ParserFactory().create(std="f2008")
+            self._fparser_tree = parser(reader)
+
+        except Exception as err:
+            raise FileInfoFParserError(
+                "Failed to get fparser tree: " + str(err)
+            ) from err
+
+        # We directly call the cache saving routine here in case that the
+        # fparser tree will be modified later on.
+        if save_to_cache:
+            self._cache_save(verbose=verbose)
+
+        return self._fparser_tree
+
+    def get_psyir_node(self, verbose: bool = False) -> FileContainer:
```

Comment:
I think `get_psyir()` is what we want here - we get the PSyIR of the contents of the file described by the FileInfo.

Reply:
I think it's an issue how things are named in Psyclone.
It returns the root node of the IR, hence `_node`.
It's not directly the entire PSyIR, but you can access it with this node.

Reply:
But I simply rename it to `get_psyir()` to get the PR through :-)